File: informatics/lab3/solutions/java/Task.java
import java.io.*;

public abstract class Task {
    private final String name;
    private final String description;

    public Task(String name, String description) {
        this.name = name;
        this.description = description;
    }

    public String getName() {
        return name;
    }

    public String getDescription() {
        return description;
    }

    public abstract void run();

    public abstract String operate(String val);

    protected String read() {
        try {
            BufferedReader reader = new BufferedReader(new InputStreamReader(System.in));

            StringBuilder input = new StringBuilder();
            String line;
            while (!(line = reader.readLine()).matches("end|к|e"))
                input.append(line).append(" ");

            reader.close();

            return input.toString();
        } catch (Exception ex) {
            System.err.println(ex.getMessage());
            return "READ_ERROR";
        }
    }

    protected void write(String str) {
        try {
            BufferedWriter writer = new BufferedWriter(new OutputStreamWriter(System.out));
            writer.append(str).flush();
            writer.close();
        } catch (Exception ex) {
            System.err.println(ex.getMessage());
        }
    }
}

File: informatics/lab3py/task.py
from abc import ABC, abstractmethod

class Task(ABC):
    @abstractmethod
    def run(self):
        pass

    @abstractmethod
    def operate(self, val):
        pass

    def read(self):
        try:
            input_data = ""
            while True:
                line = input().strip()
                if line.lower() in ["end", "к", "e"]:
                    break
                input_data += line + " "

            return input_data
        except Exception as ex:
            print(ex)
            return "READ_ERROR"

    def write(self, text):
        try:
            print(f"answer: {text}")
        except Exception as ex:
            print(ex)
